internal/controller/controllers/kube_compat.go

```diff
@@ -2,19 +2,195 @@ package controllers
 
 import (
 	"context"
+	"fmt"
 
+	certtypes "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	certmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	"github.com/sirupsen/logrus"
+	netv1 "k8s.io/api/networking/v1"
 	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-const clusterVersionCRDName = "clusterversions.config.openshift.io"
+const (
+	caCommonName                     = "Assisted Installer CA"
+	caIssuerName                     = "assisted-installer-ca"
+	certificateCRDName               = "certificates.cert-manager.io"
+	certManagerCAInjectionAnnotation = "cert-manager.io/inject-ca-from"
+	clusterVersionCRDName            = "clusterversions.config.openshift.io"
+	selfSignedIssuerName             = "assisted-installer-selfsigned-ca"
+)
 
 func ServerIsOpenShift(ctx context.Context, c client.Client) (bool, error) {
-	clusterVersionCRD := apiextensionsv1.CustomResourceDefinition{}
-	err := c.Get(ctx, types.NamespacedName{Name: clusterVersionCRDName}, &clusterVersionCRD)
-	if err == nil {
-		return true, nil
+	return crdExists(ctx, c, clusterVersionCRDName)
+}
+
+func serverSupportsCertManager(ctx context.Context, c client.Client) (bool, error) {
+	return crdExists(ctx, c, certificateCRDName)
+}
+
+func crdExists(ctx context.Context, c client.Client, crdName string) (bool, error) {
+	crd := apiextensionsv1.CustomResourceDefinition{}
+	err := c.Get(ctx, types.NamespacedName{Name: crdName}, &crd)
+	return err == nil, client.IgnoreNotFound(err)
+}
+
+func newIngress(ctx context.Context, log logrus.FieldLogger, asc ASC, name string, host string, port int32) (client.Object, controllerutil.MutateFn, error) {
```

Comment:
Does `ctx context.Context, log logrus.FieldLogger` need to be in the argument list? :o

Reply:
Nope, I suppose not. I'll remove them